test/metabase/actions/test_util.clj

```diff
@@ -11,20 +11,63 @@
    [metabase.test.data.interface :as tx]
    [toucan.db :as db]))
 
-#_ {:clj-kondo/ignore [:unused-private-var]}
-(def ^:private actions-test-data
-  "This is basically the same as [[defs/test-data]] but it only includes the `categories` table for faster loading. It's
-  meant to be reloaded at the start of every test using it so tests can do destructive things against it e.g. deleting
-  rows. (With one Table it takes ~100ms/~250ms instead of ~200ms/~450ms for H2/Postgres respectively to load all the
-  data and sync it.)"
-  ;; TODO -- this is still annoyingly SLOW we need to optimize this a bit can't be waiting 200ms for every single test
-  (tx/transformed-dataset-definition
-   "actions-test-data"
-   defs/test-data
-   (fn [database-definition]
-     (update database-definition :table-definitions (fn [table-definitions]
-                                                      (filter #(= (:table-name %) "categories")
-                                                              table-definitions))))))
+(def ^:dynamic ^:private *actions-test-data-tables*
+  #{"categories"})
+
+(defn do-with-actions-test-data-tables
+  "Impl for [[with-actions-test-data-tables]]."
+  [tables thunk]
+  ;; make sure all the table names are valid so we can catch errors/typos
+  (let [valid-table-names-set (into #{}
+                                    (map :table-name)
+                                    (:table-definitions (tx/get-dataset-definition defs/test-data)))]
+    (doseq [table-name tables]
+      (assert (contains? valid-table-names-set table-name)
+              (format "Invalid table for `with-actions-test-data-tables` %s. Valid tables: %s"
+                      (pr-str table-name)
+                      (pr-str valid-table-names-set)))))
```

Comment:
This sort of thing is a nice thing to have when writing test tooling. We check that the tables you pass in to `with-actions-test-data-tables` are all valid and fail if not. This makes it impossible to write a test that you think is checking things under one condition but actually isn't because you misspelled a table name. Example:

```clj
(with-actions-test-data-tables #{"categories" "venue"}
  (with-actions-test-data
    (delete-category! 58))) ; delete Pizza which should trigger an FK constraint
```

You might think you're triggering an FK constraint by using the `venues` Table, since several venues have `category_id 58`. But the `test-data` Table name is actually `venues`, not `venue`; if we didn't have this check here we'd actually be creating a database with only the `categories` table since `venue` doesn't match anything.